pkg/mimirpb/timeseries.go

```diff
@@ -19,11 +19,14 @@ import (
 )
 
 const (
-	minPreallocatedTimeseries         = 100
-	minPreallocatedLabels             = 20
-	minPreallocatedSamplesPerSeries   = 10
-	minPreallocatedExemplarsPerSeries = 1
-	maxPreallocatedExemplarsPerSeries = 10
+	minPreallocatedTimeseries          = 100
+	minPreallocatedLabels              = 20
+	maxPreallocatedLabels              = 100
+	minPreallocatedSamplesPerSeries    = 10
+	maxPreallocatedSamplesPerSeries    = 50
```

Comment:
Not sure if this is too aggressive, it won't make a difference if this is 10x of that in terms of memory usage, but I also don't have any data to support this.


(If I was writing this code, I'd do 1024 here or something like that)

Reply:
In the Prometheus world. We typically expect 1 sample per series. When remote write is catching up we could get multiple samples. For example, assuming 4DPM, 50 samples = 11 minutes of data. It still looks a lot coming from a single WriteRequest, considering how WAL is replayed.

What's the legit case you have in mind for 1K samples for 1 series in 1 request?